utils/python/CIME/XML/env_batch.py

```diff
@@ -178,3 +183,255 @@ def cleanupnode(self, node):
         else:
             node = EnvBase.cleanupnode(self, node)
         return node
+
+    def set_batch_system(self, batchobj, batch_system_type=None):
+        if batch_system_type is not None:
```

Comment:
This looks important but I don't understand. Does this load the contents of batchobjs (which come from config_batch.xml) into env_batch.xml?